### PR TITLE
ROX-18864: Change vuln reporting feature flag to env var

### PR DESCRIPTION
**Pull request by md2119**
## Description

Change the vulnerability reporting feature flag to env var to support disabling the feature in customer deployments if needed.

## Checklist
- [ ] Investigated and inspected CI test results
- ~Unit test and regression tests added~
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes~
If any of these don't apply, please comment below.

## Testing Performed
Wait for CI

### CODE REVIEW

**Comment by theencee** [comment]
Please also update tests/feature_flag_test.go . See line 50 for example

File: pkg/env/vuln_report_enhancements.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package env
```
Comment (charmik-redhat) [inline pkg/env/vuln_report_enhancements.go:1]: We already have `pkg/env/vuln_reporting.go` . Might be good to add the env var there to this file.

**Review by charmik-redhat — APPROVED** [review]
One comment. Otherwise lgtm.

File: pkg/env/vuln_report_enhancements.go (hunk 1 above)
Comment (md2119) [reply pkg/env/vuln_report_enhancements.go:1]: I ack your comment. To avoid blocking follow-up PRs, I will address your comment in a follow-up.

**Pull request merged**